### ISSUE TITLE
Request: Ability to create multiple wishlists

### ISSUE THREAD

**Issue by OneRadRedDad** [Wingysam/Christmas-Community#61]
So a parent can create a wishlist for their child, or two people can share an account (such as grandparents who might not be as tech savvy)

**Comment by Wingysam** [comment]
This doesn't work with the model of seeing someone's wishlist by visiting /wishlists/<account name>. In the case of a child that isn't old enough to use the app, the parent can create an account for the child then use the "impersonate" feature to log in as them and add items.

In the case of grandparents, I think it might be best to create "permanent access links" that immediately log you in as a specific user, then you can give them this link somehow. In the case of my grandparents, I would text it to them so they could log in on their phone easily. In some cases, you could create a shortcut for them on their desktop, or a bookmark.

**Comment by OneRadRedDad** [comment]
Understandable about grandparents. I made a single account for my
grandparents for now, and will instruct them to include who wants each item
in the Notes field.

Re: Children-- what if that URL were reserved for the "default" wishlist
for an account, and additional lists could be in
/wishlists/<account>/<other wishlist name>/ ?

As for impersonation, does that mean a parent would need to be an admin
account? Is it possible to set it so certain accounts can only impersonate
certain other accounts?

**Comment by Wingysam** [comment]
Yes, I assumed that the parent is also the admin. So the admin should be able to create a "sublist" under the parent's account which would have no secrecy features such as not being able to see what other people have added to your wishlist or what has been pledged?

**Comment by Wingysam** [comment]
Yeah that seems like the best way to go about this. I'll need to add per-wishlist settings in addition to sublists.

**Comment by dominch** [comment]
I'm also missing some kind of classification per user like: 
- category - for home, for hobby, for big/small gift idea 
- wishlists for some occasions (christmass, birthday) 
- private entries
having everything in one bucket is too simple

**Comment by jskidie** [comment]
> This doesn't work with the model of seeing someone's wishlist by visiting /wishlists/<account name>. In the case of a child that isn't old enough to use the app, the parent can create an account for the child then use the "impersonate" feature to log in as them and add items.
> 
> In the case of grandparents, I think it might be best to create "permanent access links" that immediately log you in as a specific user, then you can give them this link somehow. In the case of my grandparents, I would text it to them so they could log in on their phone easily. In some cases, you could create a shortcut for them on their desktop, or a bookmark.

How to create a permament access link ? and is there any documentation about ?

**Comment by fdw** [comment]
I'd like to support the original request for multiple wishlists. I would like to have different wishlists for different people/groups, because I don't want to share some things with my colleagues, for example.

If you can choose with whom you share a wishlist, this might even solve #128.

**Comment by jskiddie** [comment]
@fdw why not create multiple accounts for yourself and group them with the corresponding people ? (yes this requires grouping but as noted in, which is already in consideration as per #7 )

**Comment by fdw** [comment]
That would be possible, but then I'd have to manage multiple accounts with multiple passwords and log in and out all the time. It feels a lot more comfortable (and natural) to log in only once and then manage multiple lists, moving items between lists and maybe having one list shared with no-one that serves as a feed for the other lists.

**Comment by Wingysam** [comment]
So you use Christmas Community with your coworkers? What does that look like?

**Comment by fdw** [comment]
That was just an example, but what I'd like to do with Wishlist is to collect books that could be interesting for my work and share that with (friendly) colleagues, so that they can recommend things they've read before or find interesting. But I'd still buy them myself.

But you can see how I wouldn't want to share my other lists with them 😉

**Comment by Wingysam** [comment]
Huh interesting, so you're using Christmas Community as a recommendation list? Like you add books that you think other people should read to your wishlist?

**Comment by fdw** [comment]
It's more a recommendation for me: I use it to organize ideas for my next books. I add everything that sounds interesting, and I ask people to have a look and recommend what they've read. And when I need something new, I choose from that list.

**Comment by Wingysam** [comment]
Are you able to see the books they add to your list?

**Comment by fdw** [comment]
No, but it's not so much about them adding new things. In a sense, I maintain my personal "wish" list and ask for feedback like "What do you think I should read next?"

**Comment by Wingysam** [comment]
So it sounds like we need a few concepts:

* Group: The instance admin can create groups and assign users to them.
* Wishlist: Has a Name, belongs to a User and is associated with any number of Groups. Has some settings like "can the owner see things they didn't add?". 
    * Should the User be allowed to enable the setting where they can see what everyone else added to their list? I think children should not be trusted not to "peek".
    * I think it's best for the instance admin to configure at least some of the settings. The User should probably be allowed to decide which Groups the Wishlist is associated with.
    * Maybe there should be an instance admin setting for whether or not the User can configure the Wishlist?

**Comment by fdw** [comment]
Some thoughts, just from my perspective (but you have thought a lot more about Wishlist, so please just take them as food for thought):
- The main feature (for me) would be multiple wish lists per user.
- As a user, I'd like to define who can see each of my wish lists and not depend on the instance admin to define a group. Just let me pick from the known users, that's easy for me and easier for the admin.
- I can imagine that sharing a list with not registered users (for example, using a non-obvious link) could be helpful as well, so that you can share the list before the receivers made their account.
- A setting where I can see what people added to my list would be interesting, too. And while I think this usually should be set by the user themself, I agree that there might be cases where certain users (children, like you said) should not be able to do that. Maybe a "locked-down" account that could have certain other restrictions?

**Comment by Wingysam** [comment]
I'm having trouble envisioning some of this from a user perspective. If Bob has shared multiple wishlists with me, what will that look like on the front page? Does each wishlist have a name? Can I see who it has been shared with? Can it be shared with entire groups in addition to individual users?

**Comment by fdw** [comment]
If it were me, I'd try to keep it simple for the beginning and then see what can be improved. For example, I wouldn't use groups at all for now - that's something that can be implemented independently later.

A name for a wishlist probably makes sense. Then you can see if this one is for their birthday or some other event, maybe even just a note.

Seeing who the list has been shared with is probably also helpful, because I can already see who has claimed a gift, can't I? It's definitely not something that has to be kept secret, so why not?

**Comment by dathbe** [comment]
The ability to set a list (or individual items) as private would be great too.  I keep some things as a reminder to myself that the whole fam doesn't need to see.

**Comment by Wingysam** [comment]
I think private items is out-of-scope for Christmas Community. I want to avoid complicating the UI with features that most people don't use. If we build lists that you can share with arbitrary people then you could create a private list by just not sharing it with anyone though!

**Comment by Wingysam** [comment]
My vision for sharing with multiple users/groups looks like this:

# Wishlist Name Here
Shared with (group icon) (group icon) (group icon) | (user icon) (user icon) | +

Hovering over the icons would show the name of the group/user, clicking them would take you to a page showing info about the group or a list of the user's wishlists that you have access to.

**Comment by Wingysam** [comment]
I'd also like to have a "group invite link" sort of like Discord invite links where it lets you join the group. If you already have an account, it should ask you which wishlists you would like to share with the group. If you don't have an account, it should (with administrator allowing it) let you create an account and set your own name.
